coprocessor/fhevm-engine/host-listener/src/cmd/mod.rs

```diff
@@ -853,10 +896,74 @@ async fn db_insert_block(
     }
 }
 
+/// Runs the catchup-only loop mode **forever**.
+///
+/// This function processes blocks from `start_at_block` to `stop_at_block`,
+/// sleeps for `catchup_sleep_secs`, then repeats indefinitely.
+/// It only returns `Err` on connection failures; otherwise it loops forever.
+async fn run_catchup_only_loop(
+    args: &Args,
+    log_iter: &mut InfiniteLogIter,
+    db: &mut Database,
+    chain_id: ChainId,
+    acl_contract_address: &Option<Address>,
+    tfhe_contract_address: &Option<Address>,
+) -> anyhow::Result<()> {
+    info!("Starting catchup-only loop mode");
+
+    loop {
+        log_iter.new_log_stream(true).await;
+
+        // Process blocks until end_at_block is reached
+        while let Some(block_logs) = log_iter.next().await {
+            let _ = db_insert_block(
+                chain_id,
+                db,
+                &block_logs,
+                acl_contract_address,
+                tfhe_contract_address,
+            )
+            .await;
+        }
+
```

Comment:
this duplicates the main body. add a if !catchup loop break here and it can be fused with the main function.